dig_test.go

```diff
@@ -687,17 +687,87 @@ func TestEndToEndSuccess(t *testing.T) {
 		}))
 	})
 
+	t.Run("As with Group", func(t *testing.T) {
+		c := digtest.New(t)
+		expectedStrs := []string{"foo", "bar"}
+		for _, s := range expectedStrs {
+			s := s
+			c.RequireProvide(func() *bytes.Buffer {
+				return bytes.NewBufferString(s)
+			}, dig.Group("readers"), dig.As(new(io.Reader)))
+		}
+
+		type in struct {
+			dig.In
+			Readers []io.Reader `group:"readers"`
+		}
+
+		var actualStrs []string
+
+		c.RequireInvoke(func(got in) {
+			require.Len(t, got.Readers, 2)
+			for _, r := range got.Readers {
+				buf := make([]byte, 3)
+				_, err := r.Read(buf)
+				require.NoError(t, err)
+				actualStrs = append(actualStrs, string(buf))
+			}
+		})
+
+		assert.ElementsMatch(t, actualStrs, expectedStrs, "list of strings provided must match")
+	})
+
+	t.Run("multiple As with Group", func(t *testing.T) {
+		c := digtest.New(t)
+		expectedStrs := []string{"foo", "bar"}
+		for _, s := range expectedStrs {
+			s := s
+			c.RequireProvide(func() *bytes.Buffer {
+				return bytes.NewBufferString(s)
+			}, dig.Group("buffs"), dig.As(new(io.Reader), new(io.Writer)))
+		}
+
+		type in struct {
+			dig.In
+			Readers []io.Reader `group:"buffs"`
+			Writers []io.Writer `group:"buffs"`
+		}
+
+		var actualStrs []string
+
+		c.RequireInvoke(func(got in) {
+			require.Len(t, got.Readers, 2)
+			for _, r := range got.Readers {
+				buf := make([]byte, 3)
+				_, err := r.Read(buf)
+				require.NoError(t, err)
+				actualStrs = append(actualStrs, string(buf))
+			}
+			require.Len(t, got.Writers, 2)
+		})
+
+		assert.ElementsMatch(t, actualStrs, expectedStrs, "list of strings provided must match")
+	})
+
 	t.Run("As same interface", func(t *testing.T) {
 		c := digtest.New(t)
 		c.RequireProvide(func() io.Reader {
 			panic("this function should not be called")
```

Comment:
i know this panic isn't from your PR, but let's clean this up with t.Fatal while we're at it.